src/hal/components/toggle2nist.comp

```diff
@@ -4,61 +4,68 @@ description
 """
 Toggle2nist can be used with a momentary push button 
 to control a device that has separate on and off inputs
-and an is-on output. 
+and an is-on output.
+A debounce delay in cycles can be set. (default = 2)
 
 \[bu] On a rising edge on pin \\fIin\\fR when \\fIis-on\\fR is low: It sets \\fIon\\fR until \\fIis-on\\fR becomes high.
 
 \[bu] On a rising edge on pin \\fIin\\fR when \\fIis-on\\fR is high: It sets \\fIoff\\fR until \\fIis-on\\fR becomes low.
 
 
-       ┐     ┌─────xxxxxxxxxxxx┐           ┌─────xxxxxxxxxxxx┐
+       ┐     ┌─────xxxxxxxxxxxx┐           ┌─────xxxxxxxxxxxx┐
 .br
-in   : └─────┘     xxxxxxxxxxxx└───────────┘     xxxxxxxxxxxx└─────
+in   : └─────┘     xxxxxxxxxxxx└───────────┘     xxxxxxxxxxxx└─────
 .br
-       ┐     ┌───────────┐
+       ┐     ┌───────────┐
 .br
-on   : └─────┘           └─────────────────────────────────────────
+on   : └─────┘           └─────────────────────────────────────────
 .br
-       ┐                                   ┌───────────┐
+       ┐                                   ┌───────────┐
 .br
-off  : └───────────────────────────────────┘           └───────────
+off  : └───────────────────────────────────┘           └───────────
 .br
-       ┐                 ┌─────────────────────────────┐
+       ┐                 ┌─────────────────────────────┐
 .br
-is-on: └─────────────────┘                             └───────────
+is-on: └─────────────────┘                             └───────────
 
 """;
 
 pin in  bit in;
 pin in  bit is_on;
 pin out bit on;
 pin out bit off;
-variable int old_in;
-variable int to_state=0;
+param rw u32 debounce = 2 "debounce delay in cycles";
+variable int debounce_cntr;
+variable int state;
+
 function _ nofp;
 license "GPL";
-author "Anders Wallin";
+author "David Mueller, Based on work by Anders Wallin and John Kasunich";
```

Comment:
Isn't Anders Wallin still the author? :thinking:

Reply:
I'm not really keen on having my name there. The reason I changed that is that I believe the 'author' is the person who could be contacted when questions arise. Since I completely changed the function it seems pointless to contact Anders Wallin.

To be honest, it has always been a mystery to me as to who the 'author' is supposed to be for something that is substantially changed. Is the original author still the 'author' if somebody changes half of the content?

Reply:
I don't care, but I think maybe just a list of authors, like  `author "Anders Wallin, David Mueller";` ?